pkg/pull/signature.go

```diff
@@ -48,16 +68,16 @@ func VerifySignature(license *kotsv1beta1.License) error {
 		return errors.New("unknown global key")
 	}
 
-	if err := verify([]byte(signature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
+	if err := verify([]byte(innerSignature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
 		return errors.Wrap(err, "failed to verify key signature")
 	}
 
-	licenseMessage, err := getMessageFromLicense(license)
+	licenseMessage, err := base64.StdEncoding.DecodeString(outerSignature.LicenseData)
 	if err != nil {
-		return errors.Wrap(err, "failed to convert license to message")
+		return errors.New("failed to decode license data from outer signature")
 	}
 
-	if err := verify(licenseMessage, signature.LicenseSignature, []byte(signature.PublicKey)); err != nil {
+	if err := verify(licenseMessage, innerSignature.LicenseSignature, []byte(innerSignature.PublicKey)); err != nil {
 		return errors.Wrap(err, "failed to verify license signature")
 	}
 
```

Comment:
before returning here you should parse `licenseMessage` as a LicenseSpec
and then compare it to the other fields within the passed license's LicenseSpec (everything besides 'signature')